server/konnectors/edf.coffee

```diff
@@ -0,0 +1,1321 @@
+
+                doc.costsByCategory.standing = mee.standingCharge
+                data.consumptionStatementByMonth[mee.month] = doc
+
+                return doc
+
+            # Convenient structure to enhance data later.
+            data.consumptionStatementByYear = {}
+
+            statements = statements.concat obj.yearlyElecEnergies.map (yee) ->
+                doc =
+                    docTypeVersion: K.docTypeVersion
+                    contractNumber: data.contract.number
+                    start: yee.beginDay
+                    end: yee.endDay
+                    value: yee.consumption.energy
+                    statementType: 'estime'
+                    statementCategory: 'edelia'
+                    statementReason: 'EdeliaYearlyElecConsumption'
+                    period: yee.year
+                    cost: yee.totalCost
+                    costsByCategory: yee.consumption.costsByTariffHeading
+                    valuesByCatergory: yee.consumption.energiesByTariffHeading
+
+                doc.costsByCategory.standing = yee.standingCharge
+
+                # Add to a convenient structure to enhance them with comparisons
+                data.consumptionStatementByYear[yee.year] = doc
+                return doc
+
+            if statements.length isnt 0
+                entries.consumptionstatement = entries
+                    .consumptionstatement.concat statements
+
+            K.logger.info 'Fetched fetchEdeliaMonthlyElecConsumptions'
+
+        catch e
+            error = e
+        finally
+            callback error
+
+
+fetchEdeliaSimilarHomeYearlyElecComparisions = (
+requiredFields, entries, data, callback) ->
+    return callback() if data.noEdelia or data.noElec
+
+    K.logger.info "fetchEdeliaSimilarHomeYearlyElecComparisions"
+    getEdelia data.edeliaToken
+    , "/sites/-/similar-home-yearly-elec-comparisons?begin-year=2012"
+    , (err, response, objs) ->
+        error = null
+        try
+            if response.statusCode is 404 or response.statusCode is 500
+                K.logger.warn 'No EdeliaSimilarHomeYearlyElecComparisions'
+                data.noElec = true
+                throw null
+            if err
+                K.logger.error(
+                    'While fetchEdeliaSimilarHomeYearlyElecComparisions')
+                K.logger.error err
+                throw err
+
+            objs.forEach (obj) ->
+                # if ...
+                statement = data.consumptionStatementByYear[obj.year]
+                statement.similarHomes =
+                    site: obj.energies.site
+                    average: obj.energies.similarHomes.SH_AVERAGE_CONSUMING
+                    least: obj.energies.similarHomes.SH_LEAST_CONSUMING
+
+            K.logger.info 'Fetched fetchEdeliaSimilarHomeYearlyElecComparisions'
+        catch e
+            error = e
+        finally
+            callback error
+
+        delete data.consumptionStatementByYear
```

Comment:
Unreachable code. Or race condition ad the callback is called inside the finally